src/components/Header/components/LocationSearch/components/SearchedLocation/SearchedLocation.jsx

```diff
@@ -6,26 +6,58 @@ import type {
   SearchedLocationOwnPropsType
 } from './SearchedLocationPropsType';
 import { connect } from 'react-redux';
-import { setCurrentLocation } from '../../../../../../actions/locationsManagerActions';
+import {
+  setCurrentLocation,
+  setFavoriteCities
+} from '../../../../../../actions/locationsManagerActions';
+import { selectFavoriteCitiesId } from '../../../../../../selectors/selectorsFavorite';
+import { ReactComponent as HeartIcon } from '../../../../../../assets/img/svg/icon-heart.svg';
+
+const SearchedLocation = ({
+  location,
+  setFavoriteCities,
+  favoriteCitiesIdList,
+  ...props
+}: SearchedLocationPropsType): React$Node => {
+  const [isFavorite, setIsFavorite] = React.useState(() => {
+    return favoriteCitiesIdList.indexOf(location.id) !== -1;
+  });
+
+  const handleToggleFavorite = event => {
+    setIsFavorite(isFavorite => {
+      setFavoriteCities(location, !isFavorite);
+      return !isFavorite;
+    });
+
+    event.preventDefault();
+    event.stopPropagation();
+  };
 
-const SearchedLocation = ({ location, ...props }: SearchedLocationPropsType): React$Node => {
   return (
     <li onClick={() => props.setCurrentLocation(location)} className={classes.locationItem}>
       <a className={classes.locationItemLink} href="#">
         <div className={classes.locationName}>{location.name}</div>
         <div className={classes.locationArea}>{`${location.adminArea} / ${location.country}`}</div>
       </a>
+      <a
+        href="#"
+        className={[classes.favoriteBtn, isFavorite && classes.active].join(' ')}
```

Comment:
if isFavorite  = false you will most likely have false as a className. instead:
`[classes.favoriteBtn, isFavorite ? classes.active : ''].join(' ')`